wandb/sdk/interface/artifacts.py

```diff
@@ -986,7 +986,14 @@ def get_artifacts_cache() -> ArtifactsCache:
 
 def get_staging_dir() -> FilePathStr:
     path = os.path.join(env.get_data_dir(), "artifacts", "staging")
-    mkdir_exists_ok(path)
+    try:
+        mkdir_exists_ok(path)
+    except OSError as e:
+        raise PermissionError(
+            f"Unable to write staging files to {path}. "
+            "Set WANDB_DATA_DIR to a directory you have write access to."
```

Comment:
maybe we can use the `wandb.env` variable instead of hard-coding the name?

what do you think about this small re-phrasing?
```
Cannot write staging files in {path} due to lack of write permission. To fix this problem, please set  {env.DATA_DIR}  to a directory where you have the necessary write access.
```